File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        min_size = float('inf')
        current_sum = 0
        start = 0

        for end in range(n):
            current_sum += nums[end]
            
            while current_sum >= target:
                min_size = min(min_size, end - start + 1)
                current_sum -= nums[start]
                start += 1

        return min_size if min_size != float('inf') else 0

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.scala
object Solution {
      def minSubArrayLen(target: Int, nums: Array[Int]): Int = {

      var current_sum = 0
      var start = 0
      var min_size = Int.MaxValue

      for(end<-nums.indices){

        current_sum = current_sum+nums(end)

        while(current_sum>=target){
          min_size = Math.min(min_size, (end-start+1))
          current_sum = current_sum - nums(start)
          start+=1
        }

      }

      if(min_size==Int.MaxValue){
        return 0
      }
    else{
        return min_size
      }

  }
}